compiling on OSX 10.13 fails

As per the talk https://media.ccc.de/v/lac2018-38-mruby_zest_a_scriptable_audio_gui_framework mruby-zest should compile on OSX.

I have cloned the repo and did `git submodule init` and `git submodule update`. `make setup` and `make builddep` complete successfully.

Running `make` I get the following output (and eventual error). (I'm including the full output for documentation purposes)

```
make
ruby ./rebuild-fcache.rb
length(IR) = 198
Total time is 323.192 ms
cd deps/nanovg/src   && cc nanovg.c -c -fPIC
ar rc deps/libnanovg.a deps/nanovg/src/*.o
cd deps/pugl         && ./waf configure --no-cairo --static
Setting top to                           : /Users/fischer/work/ui/mruby-zest-build/deps/pugl
Setting out to                           : /Users/fischer/work/ui/mruby-zest-build/deps/pugl/build
Checking for 'gcc' (c compiler)          : /usr/bin/gcc

Global Configuration
 * Install prefix                        : /usr/local
 * Debuggable build                      : False
 * Build documentation                   : False

Pugl Configuration
 * OpenGL support                        : True
 * Cairo support                         : False
 * Verbose console output                : False
 * Static library                        : True
 * Unit tests                            : False

'configure' finished successfully (0.061s)
cd deps/pugl         && ./waf
Waf: Entering directory `/Users/fischer/work/ui/mruby-zest-build/deps/pugl/build'
[1/6] subst: pugl.pc.in -> build/pugl-0.pc
[2/6] c: pugl/pugl_osx.m -> build/pugl/pugl_osx.m.2.o
[3/6] c: pugl/pugl_osx.m -> build/pugl/pugl_osx.m.3.o
../pugl/pugl_osx.m:239:27: warning: 'NSShiftKeyMask' is deprecated: first deprecated in macOS 10.12 [-Wdeprecated-declarations]
        mods |= (modifierFlags & NSShiftKeyMask)     ? PUGL_MOD_SHIFT : 0;
                                 ^~~~~~~~~~~~~~
                                 NSEventModifierFlagShift
/System/Library/Frameworks/AppKit.framework/Headers/NSEvent.h:182:35: note: 'NSShiftKeyMask' has been explicitly marked deprecated here
static const NSEventModifierFlags NSShiftKeyMask              NS_DEPRECATED_WITH_REPLACEMENT_MAC("NSEventModifierFlagShift", 10.0, 10.12) = NSEventModifierFlagShift;
                                  ^
../pugl/pugl_osx.m:240:27: warning: 'NSControlKeyMask' is deprecated: first deprecated in macOS 10.12 [-Wdeprecated-declarations]
        mods |= (modifierFlags & NSControlKeyMask)   ? PUGL_MOD_CTRL  : 0;
                                 ^~~~~~~~~~~~~~~~
                                 NSEventModifierFlagControl
/System/Library/Frameworks/AppKit.framework/Headers/NSEvent.h:183:35: note: 'NSControlKeyMask' has been explicitly marked deprecated here
static const NSEventModifierFlags NSControlKeyMask            NS_DEPRECATED_WITH_REPLACEMENT_MAC("NSEventModifierFlagControl", 10.0, 10.12) = NSEventModifierFlagControl;
                                  ^
../pugl/pugl_osx.m:241:27: warning: 'NSAlternateKeyMask' is deprecated: first deprecated in macOS 10.12 [-Wdeprecated-declarations]
        mods |= (modifierFlags & NSAlternateKeyMask) ? PUGL_MOD_ALT   : 0;
                                 ^~~~~~~~~~~~~~~~~~
                                 NSEventModifierFlagOption
/System/Library/Frameworks/AppKit.framework/Headers/NSEvent.h:184:35: note: 'NSAlternateKeyMask' has been explicitly marked deprecated here
static const NSEventModifierFlags NSAlternateKeyMask          NS_DEPRECATED_WITH_REPLACEMENT_MAC("NSEventModifierFlagOption", 10.0, 10.12) = NSEventModifierFlagOption;
                                  ^
../pugl/pugl_osx.m:242:27: warning: 'NSCommandKeyMask' is deprecated: first deprecated in macOS 10.12 [-Wdeprecated-declarations]
        mods |= (modifierFlags & NSCommandKeyMask)   ? PUGL_MOD_SUPER : 0;
                                 ^~~~~~~~~~~~~~~~
                                 NSEventModifierFlagCommand
/System/Library/Frameworks/AppKit.framework/Headers/NSEvent.h:185:35: note: 'NSCommandKeyMask' has been explicitly marked deprecated here
static const NSEventModifierFlags NSCommandKeyMask            NS_DEPRECATED_WITH_REPLACEMENT_MAC("NSEventModifierFlagCommand", 10.0, 10.12) = NSEventModifierFlagCommand;
                                  ^
../pugl/pugl_osx.m:600:20: warning: 'NSClosableWindowMask' is deprecated: first deprecated in macOS 10.12 [-Wdeprecated-declarations]
                unsigned style = NSClosableWindowMask | NSTitledWindowMask;
                                 ^~~~~~~~~~~~~~~~~~~~
                                 NSWindowStyleMaskClosable
/System/Library/Frameworks/AppKit.framework/Headers/NSWindow.h:956:32: note: 'NSClosableWindowMask' has been explicitly marked deprecated here
static const NSWindowStyleMask NSClosableWindowMask NS_DEPRECATED_WITH_REPLACEMENT_MAC("NSWindowStyleMaskClosable", 10.0, 10.12) = NSWindowStyleMaskClosable;
                               ^
../pugl/pugl_osx.m:600:43: warning: 'NSTitledWindowMask' is deprecated: first deprecated in macOS 10.12 [-Wdeprecated-declarations]
                unsigned style = NSClosableWindowMask | NSTitledWindowMask;
                                                        ^~~~~~~~~~~~~~~~~~
                                                        NSWindowStyleMaskTitled
/System/Library/Frameworks/AppKit.framework/Headers/NSWindow.h:955:32: note: 'NSTitledWindowMask' has been explicitly marked deprecated here
static const NSWindowStyleMask NSTitledWindowMask NS_DEPRECATED_WITH_REPLACEMENT_MAC("NSWindowStyleMaskTitled", 10.0, 10.12) = NSWindowStyleMaskTitled;
                               ^
../pugl/pugl_osx.m:602:13: warning: 'NSResizableWindowMask' is deprecated: first deprecated in macOS 10.12 [-Wdeprecated-declarations]
                        style |= NSResizableWindowMask;
                                 ^~~~~~~~~~~~~~~~~~~~~
                                 NSWindowStyleMaskResizable
/System/Library/Frameworks/AppKit.framework/Headers/NSWindow.h:958:32: note: 'NSResizableWindowMask' has been explicitly marked deprecated here
static const NSWindowStyleMask NSResizableWindowMask NS_DEPRECATED_WITH_REPLACEMENT_MAC("NSWindowStyleMaskResizable", 10.0, 10.12) = NSWindowStyleMaskResizable;
                               ^
../pugl/pugl_osx.m:677:54: warning: 'NSAnyEventMask' is deprecated: first deprecated in macOS 10.12 [-Wdeprecated-declarations]
                                            nextEventMatchingMask: NSAnyEventMask];
                                                                   ^~~~~~~~~~~~~~
                                                                   NSEventMaskAny
/System/Library/Frameworks/AppKit.framework/Headers/NSEvent.h:158:26: note: 'NSAnyEventMask' has been explicitly marked deprecated here
static const NSEventMask NSAnyEventMask                 NS_DEPRECATED_WITH_REPLACEMENT_MAC("NSEventMaskAny", 10.0, 10.12) = NSUIntegerMax;
                         ^
../pugl/pugl_osx.m:690:55: warning: 'NSAnyEventMask' is deprecated: first deprecated in macOS 10.12 [-Wdeprecated-declarations]
                                                    nextEventMatchingMask: NSAnyEventMask];
                                                                           ^~~~~~~~~~~~~~
                                                                           NSEventMaskAny
/System/Library/Frameworks/AppKit.framework/Headers/NSEvent.h:158:26: note: 'NSAnyEventMask' has been explicitly marked deprecated here
static const NSEventMask NSAnyEventMask                 NS_DEPRECATED_WITH_REPLACEMENT_MAC("NSEventMaskAny", 10.0, 10.12) = NSUIntegerMax;
                         ^
9 warnings generated.
../pugl/pugl_osx.m:239:27: warning: 'NSShiftKeyMask' is deprecated: first deprecated in macOS 10.12 [-Wdeprecated-declarations]
        mods |= (modifierFlags & NSShiftKeyMask)     ? PUGL_MOD_SHIFT : 0;
                                 ^~~~~~~~~~~~~~
                                 NSEventModifierFlagShift
/System/Library/Frameworks/AppKit.framework/Headers/NSEvent.h:182:35: note: 'NSShiftKeyMask' has been explicitly marked deprecated here
static const NSEventModifierFlags NSShiftKeyMask              NS_DEPRECATED_WITH_REPLACEMENT_MAC("NSEventModifierFlagShift", 10.0, 10.12) = NSEventModifierFlagShift;
                                  ^
../pugl/pugl_osx.m:240:27: warning: 'NSControlKeyMask' is deprecated: first deprecated in macOS 10.12 [-Wdeprecated-declarations]
        mods |= (modifierFlags & NSControlKeyMask)   ? PUGL_MOD_CTRL  : 0;
                                 ^~~~~~~~~~~~~~~~
                                 NSEventModifierFlagControl
/System/Library/Frameworks/AppKit.framework/Headers/NSEvent.h:183:35: note: 'NSControlKeyMask' has been explicitly marked deprecated here
static const NSEventModifierFlags NSControlKeyMask            NS_DEPRECATED_WITH_REPLACEMENT_MAC("NSEventModifierFlagControl", 10.0, 10.12) = NSEventModifierFlagControl;
                                  ^
../pugl/pugl_osx.m:241:27: warning: 'NSAlternateKeyMask' is deprecated: first deprecated in macOS 10.12 [-Wdeprecated-declarations]
        mods |= (modifierFlags & NSAlternateKeyMask) ? PUGL_MOD_ALT   : 0;
                                 ^~~~~~~~~~~~~~~~~~
                                 NSEventModifierFlagOption
/System/Library/Frameworks/AppKit.framework/Headers/NSEvent.h:184:35: note: 'NSAlternateKeyMask' has been explicitly marked deprecated here
static const NSEventModifierFlags NSAlternateKeyMask          NS_DEPRECATED_WITH_REPLACEMENT_MAC("NSEventModifierFlagOption", 10.0, 10.12) = NSEventModifierFlagOption;
                                  ^
../pugl/pugl_osx.m:242:27: warning: 'NSCommandKeyMask' is deprecated: first deprecated in macOS 10.12 [-Wdeprecated-declarations]
        mods |= (modifierFlags & NSCommandKeyMask)   ? PUGL_MOD_SUPER : 0;
                                 ^~~~~~~~~~~~~~~~
                                 NSEventModifierFlagCommand
/System/Library/Frameworks/AppKit.framework/Headers/NSEvent.h:185:35: note: 'NSCommandKeyMask' has been explicitly marked deprecated here
static const NSEventModifierFlags NSCommandKeyMask            NS_DEPRECATED_WITH_REPLACEMENT_MAC("NSEventModifierFlagCommand", 10.0, 10.12) = NSEventModifierFlagCommand;
                                  ^
../pugl/pugl_osx.m:600:20: warning: 'NSClosableWindowMask' is deprecated: first deprecated in macOS 10.12 [-Wdeprecated-declarations]
                unsigned style = NSClosableWindowMask | NSTitledWindowMask;
                                 ^~~~~~~~~~~~~~~~~~~~
                                 NSWindowStyleMaskClosable
/System/Library/Frameworks/AppKit.framework/Headers/NSWindow.h:956:32: note: 'NSClosableWindowMask' has been explicitly marked deprecated here
static const NSWindowStyleMask NSClosableWindowMask NS_DEPRECATED_WITH_REPLACEMENT_MAC("NSWindowStyleMaskClosable", 10.0, 10.12) = NSWindowStyleMaskClosable;
                               ^
../pugl/pugl_osx.m:600:43: warning: 'NSTitledWindowMask' is deprecated: first deprecated in macOS 10.12 [-Wdeprecated-declarations]
                unsigned style = NSClosableWindowMask | NSTitledWindowMask;
                                                        ^~~~~~~~~~~~~~~~~~
                                                        NSWindowStyleMaskTitled
/System/Library/Frameworks/AppKit.framework/Headers/NSWindow.h:955:32: note: 'NSTitledWindowMask' has been explicitly marked deprecated here
static const NSWindowStyleMask NSTitledWindowMask NS_DEPRECATED_WITH_REPLACEMENT_MAC("NSWindowStyleMaskTitled", 10.0, 10.12) = NSWindowStyleMaskTitled;
                               ^
../pugl/pugl_osx.m:602:13: warning: 'NSResizableWindowMask' is deprecated: first deprecated in macOS 10.12 [-Wdeprecated-declarations]
                        style |= NSResizableWindowMask;
                                 ^~~~~~~~~~~~~~~~~~~~~
                                 NSWindowStyleMaskResizable
/System/Library/Frameworks/AppKit.framework/Headers/NSWindow.h:958:32: note: 'NSResizableWindowMask' has been explicitly marked deprecated here
static const NSWindowStyleMask NSResizableWindowMask NS_DEPRECATED_WITH_REPLACEMENT_MAC("NSWindowStyleMaskResizable", 10.0, 10.12) = NSWindowStyleMaskResizable;
                               ^
../pugl/pugl_osx.m:677:54: warning: 'NSAnyEventMask' is deprecated: first deprecated in macOS 10.12 [-Wdeprecated-declarations]
                                            nextEventMatchingMask: NSAnyEventMask];
                                                                   ^~~~~~~~~~~~~~
                                                                   NSEventMaskAny
/System/Library/Frameworks/AppKit.framework/Headers/NSEvent.h:158:26: note: 'NSAnyEventMask' has been explicitly marked deprecated here
static const NSEventMask NSAnyEventMask                 NS_DEPRECATED_WITH_REPLACEMENT_MAC("NSEventMaskAny", 10.0, 10.12) = NSUIntegerMax;
                         ^
../pugl/pugl_osx.m:690:55: warning: 'NSAnyEventMask' is deprecated: first deprecated in macOS 10.12 [-Wdeprecated-declarations]
                                                    nextEventMatchingMask: NSAnyEventMask];
                                                                           ^~~~~~~~~~~~~~
                                                                           NSEventMaskAny
/System/Library/Frameworks/AppKit.framework/Headers/NSEvent.h:158:26: note: 'NSAnyEventMask' has been explicitly marked deprecated here
static const NSEventMask NSAnyEventMask                 NS_DEPRECATED_WITH_REPLACEMENT_MAC("NSEventMaskAny", 10.0, 10.12) = NSUIntegerMax;
                         ^
9 warnings generated.
[4/6] cstlib: build/pugl/pugl_osx.m.3.o -> build/libpugl-0.a
[5/6] cshlib: build/pugl/pugl_osx.m.2.o -> build/libpugl-0.dylib
[6/6] vnum: build/libpugl-0.dylib -> build/libpugl-0.0.dylib build/libpugl-0.0.2.0.dylib
Waf: Leaving directory `/Users/fischer/work/ui/mruby-zest-build/deps/pugl/build'
'build' finished successfully (1.047s)
cd src/osc-bridge    && CFLAGS="-I ../../deps/libuv-v1.9.1/include " make lib
cc -I ../../deps/libuv-v1.9.1/include  -std=gnu99 -Wall -Wextra -I . -O3 -g -fPIC -c src/bridge.c src/cache.c src/parse-schema.c src/schema.c rtosc/rtosc.c
ar rcs libosc-bridge.a bridge.o cache.o parse-schema.o schema.o rtosc.o
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libosc-bridge.a(cache.o) has no symbols
cd mruby             && MRUBY_CONFIG=../build_config.rb rake
Environment is:

false
GIT   https://github.com/mruby/mgem-list.git -> build/mrbgems/mgem-list
Cloning into '/Users/fischer/work/ui/mruby-zest-build/mruby/build/mrbgems/mgem-list'...
remote: Counting objects: 303, done.
remote: Compressing objects: 100% (301/301), done.
remote: Total 303 (delta 13), reused 14 (delta 1), pack-reused 0
Receiving objects: 100% (303/303), 53.44 KiB | 0 bytes/s, done.
Resolving deltas: 100% (13/13), done.
GIT   https://github.com/iij/mruby-pack.git -> build/mrbgems/mruby-pack
Cloning into '/Users/fischer/work/ui/mruby-zest-build/mruby/build/mrbgems/mruby-pack'...
remote: Counting objects: 12, done.
remote: Compressing objects: 100% (8/8), done.
remote: Total 12 (delta 0), reused 7 (delta 0), pack-reused 0
Unpacking objects: 100% (12/12), done.
GIT   https://github.com/iij/mruby-process.git -> build/mrbgems/mruby-process
Cloning into '/Users/fischer/work/ui/mruby-zest-build/mruby/build/mrbgems/mruby-process'...
remote: Counting objects: 18, done.
remote: Compressing objects: 100% (17/17), done.
remote: Total 18 (delta 0), reused 11 (delta 0), pack-reused 0
Unpacking objects: 100% (18/18), done.
GIT   https://github.com/ksss/mruby-file-stat.git -> build/mrbgems/mruby-file-stat
Cloning into '/Users/fischer/work/ui/mruby-zest-build/mruby/build/mrbgems/mruby-file-stat'...
remote: Counting objects: 21, done.
remote: Compressing objects: 100% (18/18), done.
remote: Total 21 (delta 0), reused 8 (delta 0), pack-reused 0
Unpacking objects: 100% (21/21), done.
GIT   https://github.com/iij/mruby-errno.git -> build/mrbgems/mruby-errno
Cloning into '/Users/fischer/work/ui/mruby-zest-build/mruby/build/mrbgems/mruby-errno'...
remote: Counting objects: 15, done.
remote: Compressing objects: 100% (12/12), done.
remote: Total 15 (delta 0), reused 11 (delta 0), pack-reused 0
Unpacking objects: 100% (15/15), done.
GIT   https://github.com/iij/mruby-dir.git -> build/mrbgems/mruby-dir
Cloning into '/Users/fischer/work/ui/mruby-zest-build/mruby/build/mrbgems/mruby-dir'...
remote: Counting objects: 16, done.
remote: Compressing objects: 100% (12/12), done.
remote: Total 16 (delta 0), reused 7 (delta 0), pack-reused 0
Unpacking objects: 100% (16/16), done.
CC    src/object.c -> build/host/src/object.o
CC    src/kernel.c -> build/host/src/kernel.o
CC    src/print.c -> build/host/src/print.o
CC    src/class.c -> build/host/src/class.o
CC    src/array.c -> build/host/src/array.o
CC    src/etc.c -> build/host/src/etc.o
CC    src/state.c -> build/host/src/state.o
CC    src/init.c -> build/host/src/init.o
CC    src/vm.c -> build/host/src/vm.o
CC    src/string.c -> build/host/src/string.o
CC    src/hash.c -> build/host/src/hash.o
CC    src/codedump.c -> build/host/src/codedump.o
CC    src/debug.c -> build/host/src/debug.o
CC    src/range.c -> build/host/src/range.o
CC    src/fmt_fp.c -> build/host/src/fmt_fp.o
CC    src/dump.c -> build/host/src/dump.o
CC    src/error.c -> build/host/src/error.o
CC    src/backtrace.c -> build/host/src/backtrace.o
CC    src/numeric.c -> build/host/src/numeric.o
CC    src/enum.c -> build/host/src/enum.o
CC    src/symbol.c -> build/host/src/symbol.o
CC    src/proc.c -> build/host/src/proc.o
CC    src/pool.c -> build/host/src/pool.o
CC    src/compar.c -> build/host/src/compar.o
CC    src/version.c -> build/host/src/version.o
CC    src/crc.c -> build/host/src/crc.o
CC    src/load.c -> build/host/src/load.o
CC    src/variable.c -> build/host/src/variable.o
CC    src/gc.c -> build/host/src/gc.o
CC    mrbgems/mruby-bin-mrbc/tools/mrbc/mrbc.c -> build/host/mrbgems/mruby-bin-mrbc/tools/mrbc/mrbc.o
CC    mrbgems/mruby-compiler/core/codegen.c -> build/host/mrbgems/mruby-compiler/core/codegen.o
YACC  mrbgems/mruby-compiler/core/parse.y -> build/host/mrbgems/mruby-compiler/core/y.tab.c
CC    build/host/mrbgems/mruby-compiler/core/y.tab.c -> build/host/mrbgems/mruby-compiler/core/y.tab.o
AR    build/host/lib/libmruby_core.a
ar: creating archive /Users/fischer/work/ui/mruby-zest-build/mruby/build/host/lib/libmruby_core.a
LD    build/host/bin/mrbc
ld: library not found for -lGL
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ld: library not found for -lGL
clang: error: linker command failed with exit code 1 (use -v to see invocation)
rake aborted!
Command failed with status (1): [gcc -L"/Users/fischer/work/ui/mruby-zest-b...]
/Users/fischer/work/ui/mruby-zest-build/mruby/tasks/mruby_build_commands.rake:31:in `_run'
/Users/fischer/work/ui/mruby-zest-build/mruby/tasks/mruby_build_commands.rake:36:in `rescue in _run'
/Users/fischer/work/ui/mruby-zest-build/mruby/tasks/mruby_build_commands.rake:32:in `_run'
/Users/fischer/work/ui/mruby-zest-build/mruby/tasks/mruby_build_commands.rake:183:in `run'
/Users/fischer/work/ui/mruby-zest-build/mruby/mrbgems/mruby-bin-mrbc/mrbgem.rake:12:in `block (2 levels) in <top (required)>'
Tasks: TOP => default => all => /Users/fischer/work/ui/mruby-zest-build/mruby/build/host/lib/libmruby.flags.mak => /Users/fischer/work/ui/mruby-zest-build/mruby/build/host/lib/libmruby.a => /Users/fischer/work/ui/mruby-zest-build/mruby/build/host/mrblib/mrblib.o => /Users/fischer/work/ui/mruby-zest-build/mruby/build/host/mrblib/mrblib.c => /Users/fischer/work/ui/mruby-zest-build/mruby/build/host/bin/mrbc
(See full trace by running task with --trace)
make: *** [all] Error 1
```

It seems that the linker cannot link the GL library.
According to https://stackoverflow.com/questions/18550265/compiling-a-c-program-that-uses-opengl-in-mac-os-x the OSX frameworks should be specified instead of the GL libraries

I have dug a bit, but couldn't determine what is setup wrong. (and also, if this should compile on OSX at all)
My goal would be to try to use the GUI toolkit for AudioPlugin development (currently with https://github.com/rust-dsp/rust-vst )

Thanks for the quick answer
> As long as you run "OS=Mac make", then that erroneous linking should not be done and the build should work smoothly.

This took the build process quite a bit further, but now it fails with a missing X11 library...
```
# at this point mruby and mrbd were built and installed
#
cc -shared -o libzest.so `find mruby/build/host -type f | grep -e "\.o$" | grep -v bin` ./deps/libnanovg.a \
		./deps/libnanovg.a \
		src/osc-bridge/libosc-bridge.a \
		./deps/libuv-v1.9.1/.libs/libuv.a  -lm -lX11 -lGL -lpthread
ld: library not found for -lX11
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [all] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -18,6 +18,21 @@ all:
 		./deps/libuv-v1.9.1/.libs/libuv.a  -lm -lX11 -lGL -lpthread
 	$(CC) test-libversion.c deps/pugl/build/libpugl-0.a -ldl -o zest -lX11 -lGL -lpthread -I deps/pugl -std=gnu99
 
+osx:
+	ruby ./rebuild-fcache.rb
+	cd deps/nanovg/src   && $(CC) nanovg.c -c -fPIC
+	$(AR) rc deps/libnanovg.a deps/nanovg/src/*.o
+	cd deps/pugl         && ./waf configure --no-cairo --static
+#	cd deps/pugl         && ./waf configure --no-cairo --static --debug
+	cd deps/pugl         && ./waf
+	cd src/osc-bridge    && CFLAGS="-I ../../deps/libuv-v1.9.1/include " make lib
+	cd mruby             && MRUBY_CONFIG=../build_config.rb rake
+	$(CC) -shared -o libzest.so `find mruby/build/host -type f | grep -e "\.o$$" | grep -v bin` ./deps/libnanovg.a \
+		./deps/libnanovg.a \
+		src/osc-bridge/libosc-bridge.a \
+		./deps/libuv-v1.9.1/.libs/libuv.a  -lm -framework OpenGL -lpthread
+	$(CC) test-libversion.c deps/pugl/build/libpugl-0.a -ldl -o zest -framework OpenGL -framework AppKit -lpthread -I deps/pugl -std=gnu99
+
 windows:
 	cd deps/nanovg/src   && $(CC) -mstackrealign nanovg.c -c
 	$(AR) rc deps/libnanovg.a deps/nanovg/src/*.o

--- a/README.adoc
+++ b/README.adoc
@@ -1,3 +1,7 @@
+= Zyn-Fusion User Interface
+
+
+
 Welcome to the source of the Zyn-Fusion user interface.
 
 ### Preparations

--- a/README.adoc
+++ b/README.adoc
@@ -1,10 +1,9 @@
 = Zyn-Fusion User Interface
 
 
-
 Welcome to the source of the Zyn-Fusion user interface.
 
-### Preparations
+== Preparations
 
 After cloning the repository, initialize the submodules:
 [source,bash]
@@ -13,7 +12,7 @@ git submodule init
 git submodule update
 ----
 
-## Building
+== Building
 
 To build you'll want to use the Makefile:
 [source,bash]
@@ -24,7 +23,7 @@ make
 ----
 and the ahead-of-time qml compiler 'rebuild-fcache.rb'.
 
-### Building on OS X
+=== Building on OS X
 
 In order to make the build work on a recent OSX (at the time of writing, 10.13),
 run the make process as follows:
@@ -40,7 +39,7 @@ There will be a number of deprecation warnings, these can be ignored.
 The main difference to the Linux build are the linking options (The process links against
 the `OpenGL` and the `AppKit` frameworks.
 
-## Running
+== Running
 
 The build process will create a `libzest.so` library and the `zest` executable.
 See the link:contributing.adoc[contributing.adoc] document for information about

--- a/README.adoc
+++ b/README.adoc
@@ -1,4 +1,16 @@
 Welcome to the source of the Zyn-Fusion user interface.
+
+### Preparations
+
+After cloning the repository, initialize the submodules:
+[source,bash]
+----
+git submodule init
+git submodule update
+----
+
+## Building
+
 To build you'll want to use the Makefile:
 [source,bash]
 ----
@@ -7,6 +19,28 @@ make builddep
 make
 ----
 and the ahead-of-time qml compiler 'rebuild-fcache.rb'.
+
+### Building on OS X
+
+In order to make the build work on a recent OSX (at the time of writing, 10.13),
+run the make process as follows:
+
+[source,bash]
+----
+make setup
+make builddep
+OS=Mac make osx
+----
+There will be a number of deprecation warnings, these can be ignored.
+
+The main difference to the Linux build are the linking options (The process links against
+the `OpenGL` and the `AppKit` frameworks.
+
+## Running
+
+The build process will create a `libzest.so` library and the `zest` executable.
+See the link:contributing.adoc[contributing.adoc] document for information about
+running the application and changing it.
 
 Enjoy,
 --Mark McCurry

--- a/contributing.adoc
+++ b/contributing.adoc
@@ -95,19 +95,19 @@ First, open up 'src/mruby-zest/qml/Knob.qml'
 
 This defines a knob and it decends from the valuator base class.
 Let's take a look at the draw() function.
-Within the draw function add 'background color("ffffff")' and then save the
+Within the draw function add `background color("ffffff")` and then save the
 file.
 If hotloading is working you should see the background behind each one of the
 knobs change to white. If you aren't using hotloading, then closing and running
-"make && make run" should get you to the same spot.
+`make && make run` should get you to the same spot.
 
 From there, it's possible to probe other parts of the drawing process.
-Comment out the background change with "#" and then let's try changing the
+Comment out the background change with `#` and then let's try changing the
 color of something that is drawn.
 
 Go to the first 'vg.path' section and change the fill_color to
-'color("550000")' and in the next 'vg.path' section change the color to
-'color(:red)'.
+`color("550000")` and in the next 'vg.path' section change the color to
+`color(:red)`.
 Save the file and you should now see the new drawing changes in the live GUI.
 This is just a quick example, but you should get the idea.
 

